lib/AST/ClangTypeConverter.cpp

```diff
@@ -410,7 +410,19 @@ clang::QualType ClangTypeConverter::visitProtocolType(ProtocolType *type) {
 // Metatypes can be converted to Class when they are metatypes for concrete
 // classes. https://github.com/apple/swift/pull/27479#discussion_r344418131
 clang::QualType ClangTypeConverter::visitMetatypeType(MetatypeType *type) {
-  return getClangMetatypeType(ClangASTContext);
+  assert(type->hasRepresentation() &&
+         "metatype should have been assigned a representation");
+  switch (type->getRepresentation()) {
+  case MetatypeRepresentation::Thin:
+    return ClangASTContext.VoidTy;
```

Comment:
Does this work?  If the constructor takes a formal-in-SIL thing metatype argument, I'm surprised we can just translate it to `void`.

Reply:
Well, that's just saying that they're empty types; `TypeInfo` doesn't have a deeper meaning than that. But we can't just unconditionally turn a thin metatype into `void` if we're producing a foreign function signature; that's not legal in most positions.  Probably we need to erase the argument in the lowered type; I think we already have mechanics for that from import-as-member?

Reply:
@rjmccall I think you're right. We can't just turn these into `void`. I'm pretty sure that's what was causing the Windows bot to fail.

Reply:
I think the old implementation is actually sufficient (we'll see if it still works on a platform without objc-interop, though). The only time we should actually _use_ this path is when generating a lambda. If there's an issue with lambdas that take Meta types, I think that's a separate PR. This path is currently used when storing the converted clang type (and I think it's arguable whether we should actually be doing that at all).